src/tree/tree.scss

```diff
@@ -0,0 +1,148 @@
+@import '../style/index';
+@import '../icon/index';
+
+$iui-active-outline: thin solid t(iui-color-foreground-primary);
+$iui-child-node-spacing: calc(36px * (var(--level, 0)));
+
+@mixin iui-tree {
+  @include themed {
+    background-color: t(iui-color-background-1);
+  }
+}
+
+@mixin iui-tree-node {
+  @include iui-focus;
+  display: flex;
+  width: 100%;
+  align-items: center;
+  padding: $iui-xxs;
+  cursor: pointer;
+  box-sizing: border-box;
+  user-select: none;
+  margin-right: $iui-sm;
+  //min-height: $iui-component-height-small + (2 * $iui-xxs); //button height + padding
+  min-height: $iui-baseline * 3;
+
+  > .iui-icon {
+    @include iui-icons-default;
+    margin-right: $iui-s;
+    margin-left: $iui-sm + 1;
+  }
+
+  .iui-button + .iui-icon {
+    margin-left: $iui-xs + 1;
+  }
+
+  > .iui-button {
+    margin-right: $iui-s;
+    margin-left: $iui-s;
+
+    &:first-child {
+      margin-left: $iui-child-node-spacing;
+    }
+
+    > .iui-icon {
+      @media (prefers-reduced-motion: no-preference) {
+        transition: transform $iui-speed-fast ease-out;
+      }
+    }
+  }
+
+  > .iui-checkbox {
+    margin-left: $iui-s;
+    margin-right: $iui-child-node-spacing;
+  }
+
+  .iui-checkbox + .iui-node-label {
+    margin-left: $iui-s;
+  }
+
+  .iui-node-label {
+    min-width: 0;
+
+    &:only-child {
+      margin-left: $iui-child-node-spacing;
+    }
+  }
+
+  .iui-title,
+  .iui-caption {
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+  }
+
+  .iui-title {
+    font-size: $iui-font-size;
+  }
+
+  .iui-caption {
+    font-size: $iui-font-size-small;
+    @include themed {
+      color: t(iui-text-color-muted);
+    }
+  }
+
+  &.iui-active {
+    &,
+    &:hover {
+      @include themed {
+        background-color: rgba(t(iui-color-foreground-primary-rgb), t(iui-opacity-5));
+        outline: $iui-active-outline;
+        outline-offset: -1px;
+      }
+    }
+
+    &:focus,
+    &.iui-focused {
+      outline-width: $iui-xxs;
+      outline-offset: -2px;
+    }
+
+    &:focus:not(:focus-visible) {
+      outline: $iui-active-outline;
+      outline-offset: -1px;
+    }
+  }
+
+  &.iui-disabled {
+    &,
+    &:hover {
+      cursor: not-allowed;
+      outline: none;
+      @include themed {
+        background-color: t(iui-color-background-1);
+        color: t(iui-text-color-placeholder);
+      }
+
+      .iui-caption {
+        @include themed {
+          color: t(iui-text-color-placeholder);
+        }
+      }
+    }
+  }
+
+  &:hover {
+    @include themed {
+      background-color: rgba(t(iui-color-foreground-primary-rgb), t(iui-opacity-6));
+    }
+
+    .iui-caption {
+      @include themed {
+        color: t(iui-color-foreground-body);
+      }
+      @media (prefers-reduced-motion: no-preference) {
+        transition: color $iui-speed-fast ease;
+      }
+    }
+  }
+
+  &.iui-expanded {
+    > .iui-button > .iui-icon {
```

Comment:
Maybe we should have a separate class for expander button? I am afraid that if user for some reason will have button inside node, it might rotate its icon, too.

Reply:
Added iui-expander class

Reply:
Oh that's a good catch.  I think I agree with Vilius on this one.